velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -316,4 +316,40 @@ struct ArraySumFunction {
   }
 };
 
+template <typename T>
+struct ArrayAverageFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T)
+  template <typename TOutput, typename TInput>
```

Comment:
TOutput is always double.

Reply:
Rewrote call and callNullFree methods as non-templated.